### PR TITLE
[Base] Fixed compilation error in math.h

### PR DESCRIPTION
**Pull request by Xphalnos**
I made changes to **math.h** as these changes will be needed for a possible future migration to **C++20**.

### CODE REVIEW

File: src/xenia/base/math.h [hunk 1]
```diff
@@ -330,7 +330,7 @@ inline T rotate_right(T v, uint8_t sh) {
   constexpr unsigned char SHIFT_MASK = (CHAR_BIT * sizeof(T)) - 1;
   uint8_t rshr = sh & SHIFT_MASK;
   uint8_t lshl = static_cast<uint8_t>(-static_cast<int8_t>(sh)) & SHIFT_MASK;
-  return (n >> rshr) | (n << lshl);
+  return (v >> rshr) | (v << lshl);
```
Comment (Margen67) [inline src/xenia/base/math.h:333]: What does this have to do with C++20?

File: premake5.lua [hunk 2]
```diff
@@ -87,7 +87,7 @@ filter("configurations:Release")
     "NDEBUG",
     "_NO_DEBUG_HEAP=1",
   })
-  optimize("Speed")
+  optimize("Full")
```
Comment (Margen67) [inline premake5.lua]: What's the difference between Speed and Full? Premake documentation doesn't say much https://premake.github.io/docs/optimize
#
Edit:
For GCC Speed and Full are the same: https://github.com/premake/premake-core/blob/8c922aa6eaea5ffbbbc8e214cda947cf154f77f6/src/tools/gcc.lua#L83-L89
For Visual Studio `Speed` is `O2` while `Full` is `Ox`: https://github.com/premake/premake-core/blob/8c922aa6eaea5ffbbbc8e214cda947cf154f77f6/src/tools/msc.lua#L96-L103

This (`Ox`) is actually worse, see here: https://learn.microsoft.com/en-us/cpp/build/reference/ox-full-optimization?view=msvc-170

File: src/xenia/base/math.h (hunk 1 above)
Comment (Xphalnos) [reply src/xenia/base/math.h:333]: With `n` with C++20:
image

With `v` with C++20:
Just no errors

File: premake5.lua (hunk 2 above)
Comment (Margen67) [reply premake5.lua]: This change should be reverted.

Comment (Xphalnos) [reply premake5.lua]: I changed the title to "[MISC] Preparation for C++20", is this okay?

Comment (Gliniak) [reply premake5.lua]: Sorry, but can you change the commit name to something like: ``[Base] Fixed compilation error in math.h``

I tried porting Xenia once to C++20 for some nice features, but it is painful due to utf and more
